src/middlewares/checkPermissions.js

```diff
@@ -0,0 +1,11 @@
+import * as userServices from '../modules/user/user.services';
+
+import responseHandler from '../helpers/responseHandler';
+
+export default (permission = null) => async (req, res, next) => {
+  const { roleId } = req.body.auth;
```

Comment:
why are we passing `auth` as an object in th request body?

Reply:
The auth object contains the decoded JWT token from the authentication middleware.

Reply:
why was is appended to the req.body object? It can be confusing because I assumed it's coming with the req.body which it's not.

A more intuitive approach should be `req.auth` from the auth middleware, then destructure `roleId` from `req.auth`

Reply:
Ok. Will do that now.

Reply:
@Steelze ensure to refactor other parts of the codebase using `req.body.auth`